modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java

```diff
@@ -362,13 +376,15 @@ public ODataResponse createEntity(final PostUriInfo uriInfo, final InputStream c
 
                 ((UriInfoImpl) uriInfo).setKeyPredicates(keyPredicates);
 
-                ODataResponse response;
                 try (Connection connection = getDataSource().getConnection()) {
-                    if (entryMap != null) {
-                        response = ExpandCallBack.writeEntryWithExpand(getContext(), //
-                                (UriInfo) uriInfo, //
-                                entryMap, //
-                                contentType);
+                    updateCreateEventHandlerContext(handlerContext, connection, uriInfo, entry);
+                    ODataResponse afterCreateEntityResponse = this.odata2EventHandler.afterCreateEntity(uriInfo, requestContentType, contentType, entry, handlerContext);

```

Comment:
if you call afterCreateEntity here you should not call it again after the if statement bellow, actually you can't reach the code bellow